dataops/components/task_stream/crud.py

```diff
@@ -48,22 +49,26 @@ async def event_generator() -> str:
             """Generate event from stream."""
             request_start_time = time()
             redis_stream_offset = '0'
-            while True:
-                time_elapsed = time() - request_start_time
-                if await request.is_disconnected() or (
-                    params.request_timeout and time_elapsed >= params.request_timeout
-                ):
-                    break
-                else:
-                    redis_results = await self.streams_xread(params.session_id, redis_stream_offset)
-                    if redis_results:
-                        file_statuses = self.parse_file_status(redis_results)
-                        redis_stream_offset = file_statuses[
-                            len(file_statuses) - 1
-                        ].entry_id  # keep last entry ID as new offset
-                        filtered_file_statuses = self.filter_parsed_status(file_statuses, params)
-                        for file_status in filtered_file_statuses:
-                            yield f'{json.dumps(file_status)}'
+            try:
+                while True:
+                    time_elapsed = time() - request_start_time
+                    if await request.is_disconnected() or (
+                        params.request_timeout and time_elapsed >= params.request_timeout
+                    ):
+                        break
+                    else:
+                        await asyncio.sleep(1.0)
+                        redis_results = await self.streams_xread(params.session_id, redis_stream_offset)
+                        if redis_results:
+                            file_statuses = self.parse_file_status(redis_results)
+                            redis_stream_offset = file_statuses[
+                                len(file_statuses) - 1
+                            ].entry_id  # keep last entry ID as new offset
+                            filtered_file_statuses = self.filter_parsed_status(file_statuses, params)
+                            for file_status in filtered_file_statuses:
+                                yield f'{json.dumps(file_status)}'
+            except asyncio.CancelledError as e:
+                raise e
```

Comment:
Can we raise `UnhandledException` here and add logger?:

```suggestion
            except asyncio.CancelledError as e:
                logger.exception('SSE connection dropped')
                raise UnhandledException()
```